### PR TITLE
boards: sam_e70_xplained: Enable userspace

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
Enabled userspace support for Atmel SAMe70.

Confirmed this works on tests/kernel/mem_protect/userspace and tests/kernel/mem_protect/mem_protect.

### CODE REVIEW

**Review by jukkar — COMMENTED** [review]
Typo in commit message, s/usersapce/userspace/

**Comment by AdithyaBaglody** [comment]
@jukkar thanks for catching it

File: arch/arm/soc/atmel_sam/common/arm_mpu_regions.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+#include <soc.h>
+#include <arch/arm/cortex_m/mpu/arm_mpu.h>
+
+#include "arm_mpu_mem_cfg.h"
+
+/* System Private Peripheral Bus */
+#define PPB_BASE  0xE0000000
+
+static struct arm_mpu_region mpu_regions[] = {
```
Comment (andrewboie) [inline arch/arm/soc/atmel_sam/common/arm_mpu_regions.c]: I suspect we can drastically reduce the size of this.
Right now there is a background mapping set in the MPU which grants supervisor mode RW (no execute) access to the entire memory map, which acts as a least-precedence region for the entire address space.
We will need to keep the FLASH_0 region.
But with the background map I think we can discard RAM_0, PERIPHERAL_0, and PPB_0. Hence freeing up more regions for use with memory domains.
@agross-linaro @chunlinhan is my thinking sound on this?

File: arch/arm/soc/atmel_sam/common/arm_mpu_regions.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	/* Region 0 */
+	MPU_REGION_ENTRY("FLASH_0",
+			 CONFIG_FLASH_BASE_ADDRESS,
+			 REGION_FLASH_ATTR(REGION_FLASH_SIZE)),
+
+	/* Region 1 */
+	MPU_REGION_ENTRY("RAM_0",
+			 CONFIG_SRAM_BASE_ADDRESS,
+			 REGION_RAM_ATTR(REGION_SRAM_0_SIZE)),
```
Comment (andrewboie) [inline arch/arm/soc/atmel_sam/common/arm_mpu_regions.c]: If my above assertion is NOT correct, why don't we have a region for the REGION_SRAM1 stuff you put in the arm_mpu_mem_cfg.h? Not all memory will be able to be addressed.
This is probably moot though, I think we can just leverage the background map and get rid of this entirely.

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]

**Review by agross-oss — APPROVED** [review]
the long ifdef elif blocks are a little brute force, but i suppose it's ok.

**Comment by AdithyaBaglody** [comment]
Delta:
Fixed the extraneous memory region definitions.

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**